src/test/test-process-util.c

```diff
@@ -355,50 +355,80 @@ static void test_get_process_cmdline_harder(void) {
         _exit(0);
 }
 
-static void test_rename_process_one(const char *p, int ret) {
+static void test_rename_process_now(const char *p, int ret) {
         _cleanup_free_ char *comm = NULL, *cmdline = NULL;
-        pid_t pid;
         int r;
 
-        pid = fork();
-        assert_se(pid >= 0);
-
-        if (pid > 0) {
-                siginfo_t si;
-
-                assert_se(wait_for_terminate(pid, &si) >= 0);
-                assert_se(si.si_code == CLD_EXITED);
-                assert_se(si.si_status == EXIT_SUCCESS);
-
-                return;
-        }
-
-        /* child */
         r = rename_process(p);
-
         assert_se(r == ret ||
                   (ret == 0 && r >= 0) ||
                   (ret > 0 && r > 0));
 
         if (r < 0)
-                goto finish;
+                return;
 
 #ifdef HAVE_VALGRIND_VALGRIND_H
         /* see above, valgrind is weird, we can't verify what we are doing here */
         if (RUNNING_ON_VALGRIND)
-                goto finish;
+                return;
 #endif
 
         assert_se(get_process_comm(0, &comm) >= 0);
         log_info("comm = <%s>", comm);
         assert_se(strneq(comm, p, 15));
 
         assert_se(get_process_cmdline(0, 0, false, &cmdline) >= 0);
-        log_info("cmdline = <%s>", cmdline);
-        assert_se(strneq(p, cmdline, strlen("test-process-util")));
-        assert_se(startswith(p, cmdline));
+        /* we cannot expect cmdline to be renamed properly without privileges */
+        if (geteuid() == 0) {
+                log_info("cmdline = <%s>", cmdline);
+                assert_se(strneq(p, cmdline, strlen("test-process-util")));
+                assert_se(startswith(p, cmdline));
+        } else
+                log_info("cmdline = <%s> (not verified)", cmdline);
```

Comment:
Oddly enough, when I run the test as an ordinary user, I get
```
Skipping PR_SET_MM, as we don't have privileges.
comm = <one>
cmdline = <one> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <more>
cmdline = <mor> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <time!>
cmdline = <tim> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <0>
cmdline = <0> (not verified)
```
while running as root gives
```
comm = <one>
cmdline = <one>
comm = <more>
cmdline = <more>
Skipping PR_SET_MM, as we don't have privileges.
comm = <time!>
cmdline = <more> (not verified)
Skipping PR_SET_MM, as we don't have privileges.
comm = <0>
cmdline = <more> (not verified)
```
At least the test now succeeds both ways.